### PR TITLE
Skip comments in requirements and test dependencies

### PR DESCRIPTION
**Pull request by garrettgu10**
## What this PR solves / how to test

Fixes EW-8270. Revised version of #5375 ; this PR is separate because they're both done with forks. 

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: we will be documenting this in other places

### CODE REVIEW

File: packages/wrangler/e2e/dev.test.ts [hunk 1]
```diff
@@ -362,6 +362,51 @@ describe("basic dev python tests", () => {
...
+					{
+						"name": "${workerName}",
+						"version": "0.0.0",
+						"private": true
+					}
+					`,
+		}));
+	});
+
+	it("can parse requirments file and use dependencies", async () => {
```
Comment (penalosa) [inline packages/wrangler/e2e/dev.test.ts:396]:
```suggestion
	it("can parse requirements file and use dependencies", async () => {
```

File: packages/wrangler/src/deployment-bundle/find-additional-modules.ts [hunk 2]
```diff
@@ -39,6 +39,16 @@ function isValidPythonPackageName(name: string): boolean {
 	return regex.test(name);
 }
 
+/**
+ * Strips comments and surrounding whitespace from a python requirements.txt line
+ * See https://pip.pypa.io/en/latest/reference/requirements-file-format/ (Comments)
+ * @param requirement The requirement to sanitize
+ */
+function stripPythonRequirement(requirement: string): string {
```
Comment (penalosa) [inline packages/wrangler/src/deployment-bundle/find-additional-modules.ts:47]:
```suggestion
function stripComments(requirement: string): string {
```

**Review by penalosa — APPROVED** [review]
I'm a tad concerned about how fragile this might be, but let's land and see how it goes

**Comment by petebacondarwin** [comment]
I think this may have caused the e2e test failing: https://github.com/cloudflare/workers-sdk/actions/runs/8661443570/job/23752569600#step:11:129

**Pull request merged**